index.js

```diff
@@ -1,26 +1,39 @@
-const express = require('express');
-const app = express();
-const bodyParser = require('body-parser');
-
-// parse application/x-www-form-urlencoded
-app.use(bodyParser.urlencoded({ extended: false }));
-
-// parse application/json
-app.use(bodyParser.json());
-
-app.use(require('./app/controllers/searchController'));
-
-app.use(require('./app/controllers/airportDetailsController'));
-
-app.use(require('./app/controllers/flightDetailController'));
-
-app.use(require('./app/controllers/seatDetailsController'));
-
-let server = app.listen(8080);
-console.log('Server running in port: 8080 ...');
-
-module.exports = server;
-module.exports.stop = (done) => {
-    server.close();
-    done();
+const searchController = require('./app/controllers/searchController');
+const airportDetailsController = require('./app/controllers/airportDetailsController');
+const flightDetailsController = require('./app/controllers/flightDetailController');
+const seatDetailsController = require('./app/controllers/seatDetailsController');
+
+function createPromise(method) {
+    return new Promise(function(resolve, reject) {
+        method((err, result) => {
+            if (err) {
+                console.log(err);
+                reject(Error(err));
+            } else {
+                resolve(result);
+            }
+        });
+    });
+}
+
+exports.handler = async (event) => {
+    // if there is nothing in querystring, then just pass an empty obj instead of undefined
+    const filter = (event && event.params && event.params.querystring) ? event.params.querystring : {};
+    const path = event.context["resource-path"];
```

Comment:
DeepScan made a good point about lines 21 and 22: in line 21 you check whether `event` is null, but then in line 22 you access a property on it without any such check. Either the check in line 21 is unnecessary, or this access should also be guarded by a null check.